analyzers/src/SonarAnalyzer.Common/Helpers/HashCode.cs

```diff
@@ -28,12 +28,16 @@ public static class HashCode    // Replacement for System.HashCode that is avail
         private const uint PreMultiplier = 3266489917U;
         private const uint PostMultiplier = 668265263U;
         private const int RotateOffset = 17;
+        private const int IntSeed = 393241;
 
         public static int DictionaryContentHash<TKey, TValue>(IDictionary<TKey, TValue> dictionary) =>
             dictionary.Aggregate(0, (seed, kvp) => seed ^ Combine(kvp.Key, kvp.Value));
 
-        public static int EnumerableContentHash<TValue>(IEnumerable<TValue> enumerable) =>
-            enumerable.Aggregate(0, (seed, x) => Combine(seed, x));
+        public static int EnumerableUnorderedContentHash<TValue>(IEnumerable<TValue> enumerable) =>
+            enumerable.Aggregate(IntSeed, (seed, x) => seed ^ (x?.GetHashCode() ?? 0));
+
+        public static int EnumerableOrderedContentHash<TValue>(IEnumerable<TValue> enumerable) =>
+            enumerable.Aggregate(IntSeed, Combine);
```

Comment:
For these two, I'd consider single-line documentation comments that will explain the different usage.